src/core/net/ip6_address.hpp

```diff
@@ -917,10 +919,7 @@ class Address : public otIp6Address, public Equatable<Address>, public Clearable
      * @retval false  The IPv6 address is larger than or equal to @p aOther.
      *
      */
-    bool operator<(const Ip6::Address &aOther) const
-    {
-        return memcmp(mFields.m8, aOther.mFields.m8, sizeof(Ip6::Address)) < 0;
-    }
+    bool operator<(const Address &aOther) const { return memcmp(mFields.m8, aOther.mFields.m8, sizeof(Address)) < 0; }
```

Comment:
Can we compare `m32` directly ?

```suggestion
    bool operator<(const Address &aOther) const { return BigEndian::HostSwap32(mFields.m32) < BigEndian::HostSwap32(aOther.mFields.m32) }
```

Reply:
@simonlingoogle , this change is actually in `Ip6::Address` not `Ip4::Address`. The IPv6 has more than one `m32[]` field, so `memcmp` is a better way to compare all bytes.

The main change here in this commit/PR is just replacing `Ip6::Address` with `Address` as the type used in the method declaration (since it is all already under the namespace `Ip6`).
```
bool operator<(const Ip6::Address &aOther) const
``` 
to  
```
bool operator<(const Address &aOther) const
```

Reply:
Thanks for the tip.